components/service/glean-gradle-plugin/src/main/groovy/mozilla/components/service/glean-gradle-plugin/GleanGradlePlugin.groovy

```diff
@@ -0,0 +1,287 @@
+                variant.registerJavaGeneratingTask(generateKotlinAPI, new File(sourceOutputDir))
+            }
+        }
+    }
+
+    void apply(Project project) {
+        // This installs a Miniconda3 environment into the gradle user home directory
+        // so that it will be shared between all libraries that use Glean.  This is
+        // important because it is approximately 300MB in installed size.
+        File condaBootstrapDir = new File(
+            project.getGradle().gradleUserHomeDir,
+            "glean/bootstrap-${MINICONDA_VERSION}"
+        )
+        File condaDir = new File(
+            condaBootstrapDir,
+            "Miniconda3"
+        )
+
+        // Define the path to the Miniconda3 installation as a buildConfig variable so
+        // that tests that need to run JSON through our JSON schema can do so.
+        project.android {
+            defaultConfig {
+                buildConfigField(
+                    "String",
+                    "GLEAN_MINICONDA_DIR",
+                    // Carefully escape the string here so it will support `\` in
+                    // Windows paths correctly.
+                    JsonOutput.toJson(condaDir.path)
+                )
+            }
+        }
+
+        // Even though we are installing the Miniconda environment to the gradle user
+        // home directory, the gradle-python-envs plugin is hardcoded to download the
+        // installer to the project's build directory. Doing so will fail if the
+        // project's build directory doesn't already exist. This task ensures that
+        // the project's build directory exists before downloading and installing the
+        // Miniconda environment.
+        // See https://github.com/JetBrains/gradle-python-envs/issues/26
+        // The fix in the above is not actually sufficient -- we need to add createBuildDir
+        // as a dependency of Bootstrap_CONDA (where conda is installed), as the preBuild
+        // task alone isn't early enough.
+        Task createBuildDir = project.task("createBuildDir") {
+            description = "Make sure the build dir exists before creating the Python Environments"
+            onlyIf {
+                !project.file(project.buildDir).exists()
+            }
+            doLast {
+                project.logger.lifecycle("Creating build directory:" + project.buildDir.getPath())
+                project.buildDir.mkdir()
+            }
+        }
+
+        // Configure the Python environments
+        project.envs {
+            bootstrapDirectory = condaBootstrapDir
+            pipInstallOptions = "--trusted-host pypi.python.org --no-cache-dir"
+
+            // Setup a miniconda environment. conda is used because it works
+            // non-interactively on Windows, unlike the other options
+            conda "Miniconda3", "Miniconda3-${MINICONDA_VERSION}", "64", ["glean_parser==${GLEAN_PARSER_VERSION}"]
+        }
+
+        project.tasks.whenTaskAdded { task ->
```

Comment:
For later, this whole task-before-task would be easier if you added a `doFirst` to the actual `Bootstrap_CONDA` task.  But this will work too.

Reply:
We had tried that in an earlier revision (of the script this plugin is based on).  The problem is that the `Bootstrap_CONDA` task isn't created until later, so we can't just call `doFirst` on it at this point.